app/models/derived_variables/sales_log_variables.rb

```diff
@@ -17,7 +17,20 @@ def set_derived_fields!
       self.homonth = hodate.month
       self.hoyear = hodate.year
     end
-    self.deposit = value if outright_sale? && mortgage_not_used?
+
+    if outright_sale? && mortgage_not_used?
+      self.deposit = value
+    elsif outright_sale? && mortgage_use_unknown?
+      self.deposit = nil
+    elsif outright_sale? && mortgageused_changed? && !deposit_changed?
```

Comment:
This is the change that fixes the reported issue. Previously the deposit was preserved during this change, and immediately became subject to validation that it might not satisfy.